h/util/logging_filters.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Logging Filters."""
+
+import logging
+
+
+# logging levels from https://docs.python.org/2/library/logging.html#logging-levels
+LEVELS = {"CRITICAL": 50,
+          "ERROR": 40,
+          "WARNING": 30,
+          "INFO": 20,
+          "DEBUG": 10,
+          "NOTSET": 0}
+
+
+class ExceptionFilter(logging.Filter):
+    """Filter out the specified exceptions with specified logging level."""
+
+    def __init__(self, ignore_exceptions):
+        """
+        Configure filtering out of the specified exceptions with specified logging level.
+
+        ignore_exceptions: a tuple of tuples ((exception type, loglevel))
+                           example: (("requests.exceptions.ReadTimeout", "WARNING"),)
+        """
+        super(ExceptionFilter, self).__init__()
+
+        # build decoder dict where both string and int map to int logging level
```

Comment:
Ah I see, so the purpose of `logging_levels` is to allow levels in the second element of `ignore_exceptions` tuples to be int values or string names. In either case, if an unrecognised logging level or value is used, the same exception is thrown.